### PR TITLE
cmd_runner mod util: improvements

### PR DESCRIPTION
**Pull request by russoz**
##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

plugins/module_utils/cmd_runner.py

### CODE REVIEW

File: plugins/module_utils/cmd_runner.py [hunk 1]
```diff
@@ -274,6 +282,7 @@ def __init__(self, runner, args_order, output_process, ignore_value_none, check_
         self.runner = runner
         self.args_order = tuple(args_order)
         self.output_process = output_process
+        # DEPRECATION: parameter ignore_value_none at the context level is deprecated and will be removed in community.general 12.0.0
```
[the remark this answers is not in the document]
Comment (russoz) [reply plugins/module_utils/cmd_runner.py:285]: This internal class should only be instantiated through `CmdRunner.__call__()`, so if the developer is callling it directly they are subject to breaking changes like this.

File: plugins/module_utils/cmd_runner.py [hunk 2]
```diff
@@ -89,12 +89,15 @@ def __str__(self):
...
 class _ArgFormat(object):
+    # DEPRECATION: set default value for ignore_none to True in community.general 12.0.0
     def __init__(self, func, ignore_none=None, ignore_missing_value=False):
         self.func = func
         self.ignore_none = ignore_none
         self.ignore_missing_value = ignore_missing_value
 
+    # DEPRECATION: remove parameter ctx_ignore_none in community.general 12.0.0
     def __call__(self, value, ctx_ignore_none=True):
+        # DEPRECATION: replace ctx_ignore_none with True in community.general 12.0.0
```
[the remark this answers is not in the document]
Comment (russoz) [reply plugins/module_utils/cmd_runner.py:100]: This internal class' method should only be called from  `_CmdRunnerContext.run()`, so if the developer is callling it directly they are subject to breaking changes like this.

File: plugins/module_utils/cmd_runner.py [hunk 3]
```diff
@@ -246,6 +253,7 @@ def __init__(self, module, command, arg_formats=None, default_args_order=(),
     def binary(self):
         return self.command[0]
 
+    # remove parameter ignore_value_none in community.general 12.0.0
     def __call__(self, args_order=None, output_process=None, ignore_value_none=True, check_mode_skip=False, check_mode_return=None, **kwargs):
```
[the remark this answers is not in the document]
Comment (russoz) [reply plugins/module_utils/cmd_runner.py:257]: Here is the only public entrypoint that would provide the deprecated parameter, so it is the only place where that mechanism makes sense.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks for your contribution!

**Pull request merged**